### ISSUE TITLE
get retweets media 

### ISSUE THREAD

**Issue by SignCube** [Yortw/tweetmoasharp#63]
hi 
this is my code . i have no media entities for retweets . whats th problem . how can i fix it 
` var service = new TwitterService(_consumerKey, _consumerSecretKey);
            service.AuthenticateWith(_acessToken, _accessTokenSecret);
            //var currentTweets = service.ListTweetsOnSpecifiedUserTimeline(screenName: "screen name", page: 1, count: 100);
            var list = service.ListTweetsOnUserTimeline(new ListTweetsOnUserTimelineOptions()
            {
                ScreenName = "ali_persian999",
            });`

**Comment by Yortw** [comment]
Hi,

For media you should use ExtendedEntitites. The regular entities for media are basically deprecated now, and only ever contain 1 still image at most (gifs and videos only appear in extended entities, which also contains all still images from 1 - 4 if the tweet contains any).

Mainly your problem with regards to retweets is that retweets do not contain the media. The retweet just points to the original tweet, the original tweet contains the media. You can use

```c#
    tweet.RetweetedStatus.ExtendedEntities
```

to obtain the media from the retweeted status, with appropriate null checks of course.

You can check this by looking at the RawSource property of a tweet object to see the json returned. If you look at this retweet (https://twitter.com/ali_persian999/status/924655603502415872) on that timeline, you'll see the retweet json contains no extendedentities at all and only mentions in the regular entities. If you drill down to the retweeted status property, you'll find that has the attached media.

![image](https://user-images.githubusercontent.com/6725891/32687000-d470545e-c716-11e7-9cb0-8be26a6d4fe6.png)

Also note the first tweet on that timeline (https://twitter.com/ali_persian999/status/929063557802676225) does not contain any media. It contains a URL, and the web page linked to has either OpenGraph or Twitter Card metadata which many Twitter clients will use to embed the video preview of the link.

Basically this is the way Twitter returns it's data. 'TweetSharp is a wrapper around the Twitter API' meaning we don't modify of move the data around much, we try to keep it as close to what Twitter does as possible. It's a Twitter issue and working as designed. Check the Twitter docs is you need a better understanding.
